### PR TITLE
Add iterm2-tools, catimg, and imgurpython

### PR DESCRIPTION
**Pull request by asmeurer**
Note: this technically only works on OS X, because the only terminal that really implements these is iTerm2, but there's no reason another terminal couldn't implement these codes (and indeed, the defunct, FinalTerm implements the shell integration).

### CODE REVIEW

**Comment by asmeurer** [comment]
What is your policy is on adding multiple recipes, because they all depend on one another. Should they be separate PRs or the same PR? And can you add the answer to the FAQ either way.

**Comment by jakirkham** [comment]
It's fine if they are dependent on each other to add them to the same PR.

**Comment by asmeurer** [comment]
I just uploaded iterm2-tools 2.3. Is pypi.io failing because it hasn't propogated yet?

**Comment by jakirkham** [comment]
That has been known to happen before. Let's give it a bit of time and restart.

**Comment by asmeurer** [comment]
I changed the url to normal pypi.

**Comment by asmeurer** [comment]
I already know this is going to fail because catimg doesn't work in Python 2. How do I tell it otherwise? `python 3` in the build dependencies?

File: recipes/catimg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+source:
+  fn: catimg-1.1.1.tar.gz
+  url: https://pypi.python.org/packages/source/c/catimg/catimg-1.1.1.tar.gz
+  md5: 94227c65936a5e4cada9260efe0125c4
+
+build:
+  entry_points:
+    - catimg = catimg.__main__:main
+  number: 0
```
Comment (jakirkham) [inline recipes/catimg/meta.yaml]: `skip: true  # [py2k]`

Comment (jakirkham) [inline recipes/catimg/meta.yaml]: Done below.

FWIW I did included 2 spaces in that code snippet, but it makes them appear as 1. 😒

File: recipes/imgurpython/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+test:
+  imports:
+    - imgurpython
+    - imgurpython.helpers
+    - imgurpython.imgur
+    - imgurpython.imgur.models
+
+about:
+  home: https://github.com/Imgur/imgurpython
+  license: MIT License
```
Comment (jakirkham) [inline recipes/imgurpython/meta.yaml]: Just `MIT` is fine here. `License` can be dropped.

**Comment by jakirkham** [comment]
I'd recommend using Jinja for the versions. The example is a great reference for this. It really makes it easier to update packages. It also avoids silly mistakes like bumping the version, but using the same source for the old version.

**Comment by jakirkham** [comment]
Once PyPI stabilizes, which may be the case now, we should use pypi.io for all the urls.

**Comment by jakirkham** [comment]
> Note: this technically only works on OS X

If that is the case, maybe we should be adding `skip: true  # [not osx]` to the recipe(s) as relevant.

> there's no reason another terminal couldn't implement these codes (and indeed, the defunct, FinalTerm implements the shell integration). 

I see your point. It is not a crisis either way. I just don't like wasting CI time if it is going to generate something that is not used. So, if it is all the same, the skip would be nice. Once there is another terminal on another platform that works we can easily change this and we can happily walk you through the process.

File: recipes/imgurpython/meta.yaml (hunk 2 above)
Comment (asmeurer) [inline recipes/imgurpython/meta.yaml]: I guess that came from skeleton. 

I know Continuum has a set of normalized licenses that they use. Might be worth talking with them to use the same list for conda-forge (and linting against it).

**Comment by asmeurer** [comment]
OK, it's using pypi.io again. Restart when you think it will have propagated.

File: recipes/imgurpython/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/imgurpython/meta.yaml]: Yeah, linting licenses has been a long discussion ( https://github.com/conda-forge/conda-smithy/issues/88 ) and has seen at least one abandoned proposal ( https://github.com/conda-forge/conda-smithy/pull/93 ).

**Comment by jakirkham** [comment]
Thanks. Restarted.

**Comment by asmeurer** [comment]
Ah, apparently the mistake was on my end.

File: recipes/catimg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  fn: catimg-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/c/catimg/catimg-{{ version }}.tar.gz
+  md5: 94227c65936a5e4cada9260efe0125c4
+
+build:
+  entry_points:
+    - catimg = catimg.__main__:main
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip: true  # [py2k]
```
Comment (asmeurer) [inline recipes/catimg/meta.yaml]: I forgot to skip this on Linux, but it worked anyway?

File: recipes/iterm2-tools/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+source:
+  fn: iterm2-tools-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/i/iterm2-tools/iterm2-tools-{{ version }}.tar.gz
+  sha1: a49135aa73f45e135d4a27b6693e10afe597dcb6
+
+build:
+  number: 0
+  script: python setup.py install
+  skip:  # [not osx]
```
Comment (jakirkham) [inline recipes/iterm2-tools/meta.yaml]: Because our skip condition is missing `true`. 😜

Comment (jakirkham) [inline recipes/iterm2-tools/meta.yaml]: Interesting that `conda-build` doesn't give us a failure here. Normally it is very sensitive to empty dictionaries (or strings where they shouldn't be).

**Comment by asmeurer** [comment]
Why is Appveyor failing? My catimg selector should skip it as far as I can tell. It worked on CircleCI (it only built imgurpython, as it was supposed to).

**Comment by ocefpaf** [comment]
@asmeurer that is a bug in the old version of `conda-build-all` that does not know about the "final" rendered recipe in the latest `conda-build`.

That bug is fixed, but due to another bug this version never made into the channel.

TL;DR we can merge this and things should just work.

Are you done with the recipe? Can I merge this?

**Comment by jakirkham** [comment]
LGTM

**Comment by asmeurer** [comment]
Yeah, I'm done.
